test/jasmine/tests/legend_test.js

```diff
@@ -540,3 +540,73 @@ describe('legend relayout update', function() {
         });
     });
 });
+
+describe('legend orientation change:', function() {
+    'use strict';
+
+    var mock = require('@mocks/legend_horizontal_autowrap.json'),
+        gd,
+        node;
+
+    beforeEach(function(done) {
+        gd = createGraphDiv();
+        Plotly.plot(gd, mock.data, mock.layout).then(function() {
+            node = d3.select('g.legend').select('rect');
+            done();
+        });
+    });
+
+    afterEach(destroyGraphDiv);
+
+    it('horizontal with lots of items wraps traces to new lines', function(done) {
+        var nw = node.property('width').baseVal.value;
+        var gw = mock.layout.width;
+        var maxTraceHeight = 0,
+            maxTraceWidth = 0,
+            maxOffsetX = 0,
+            maxOffsetY = 0,
+            traceGap = gd._fullLayout.legend.tracegroupgap || 5;
+
+        d3.selectAll('g.legend g.traces').each(function() {
+            var b = d3.select(this).select('text').node().getBBox(),
+                w = b.width + traceGap + 40,
+                h = d3.select(this).node().getBBox().height,
+                t = d3.transform(d3.select(this).attr('transform'));
+            //console.log(t.translate[0], t.translate[1], w, h);
+            maxOffsetX = Math.max(maxOffsetX, t.translate[0]);
+            maxOffsetY = Math.max(maxOffsetY, t.translate[1]);
+            maxTraceWidth = Math.max(maxTraceWidth, w);
+            maxTraceHeight = Math.max(maxTraceHeight, h);
+        });
+
+        //legend rect less than width of plot
+        expect(nw).toBeLessThan(gw);
+        expect(maxTraceHeight).toBeGreaterThan(0);
+        expect(maxTraceWidth).toBeGreaterThan(0);
+        expect(maxOffsetX).toBeGreaterThan(0);
+        expect(maxOffsetY).toBeGreaterThan(0);
+
+        //none of the traces are pushed off the graph
+        expect(maxOffsetX).toBeLessThan(mock.layout.width);
+        expect(maxOffsetY).toBeLessThan(mock.layout.height);
+
+        done();
+    });
+
+    it('changing to vertical orientation', function(done) {
+        //change mock layout and refresh graph
+        mock.layout.legend.orientation = 'v';
+        mock.layout.legend.y = 1;
+        done();
+    });
+
+    it('changed to vertical, scrollbar appeared', function(done) {
+        //changed orientation to vertical
+        expect(gd._fullLayout.legend.orientation).toBe('v');
```

Comment:
@psalmody can you rewrite this test as 

```js
Plotly.relayout(gd, 'legend.orientation', 'v').then(function() {
  /* assertions */

 return Plotly.relayout(gd, 'legend.orientation', 'h');
}.then(function() {
  /* assertions (back to base case) /*
});
```

and remove the assertion-less `it('changing to vertical orientation', ...` above.